pkg/operator/controller/status/controller.go

```diff
@@ -355,6 +404,37 @@ func (r *reconciler) getOperatorState(ingressNamespace, canaryNamespace string,
 			}
 			state.unmanagedGatewayAPICRDNames = extension.UnmanagedGatewayAPICRDNames
 		}
+
+		if r.config.GatewayAPIControllerEnabled {
```

Comment:
I'm wondering about the logic to put the check for `Gateway`/`GatewayClass` CRDs into this condition while `Subscription` check remains in the condition above (`GatewayAPI` featuregate). `Subscription` and `Istio` CRD can only be created when `GatewayAPIController` is enabled while `Gateway`/`GatewayClass` CRDs are created when `GatewayAPI` is enabled.

Reply:
That's a fair point.  I address it with https://github.com/openshift/cluster-ingress-operator/compare/1c68435a08d7a3adac8744082ed74146548b9f7c..98b6246a365277d9feb157ece380d5b20e55332b by conditionalizing more logic on `GatewayAPIControllerEnabled`.  This also relates to (and conflicted with) the changes in #1232.